sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/cardinality/ApproximateDistinct.java

```diff
@@ -0,0 +1,352 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sketching.cardinality;
+
+import com.clearspring.analytics.stream.cardinality.CardinalityMergeException;
+import com.clearspring.analytics.stream.cardinality.HyperLogLogPlus;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.apache.beam.sdk.coders.ByteArrayCoder;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.coders.CustomCoder;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.display.DisplayData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@code PTransform}s for computing the number of distinct elements in a {@code PCollection}, or
+ * the number of distinct values associated with each key in a {@code PCollection} of {@code KV}s.
+ *
+ * <p>This class uses the HyperLogLog algorithm, and more precisely
+ * the improved version of google (HyperLogLog+).
+ * <br>The implementation comes from Addthis' library Stream-lib : https://github.com/addthis/stream-lib
```

Comment:
<!--thread_id:cc_131403647_t; commit:6bf51fdf6afe6f389415792f335dd08adf9fa85a; resolved:0-->
<!--section:context-quote-->
> **ArnaudFnr** wrote:
> Yes didn't think about that, thanks.

<!--section:body-->
I would suggest inlining the links within the text like:

```<br>The implementation comes from <a href="https://github.com/addthis/stream-lib">Addthis' Stream-lib library</a>.```

For example, you could rephrase the section linking the papers as:
```The <a href="http://algo.inria.fr/flajolet/Publications/FlFuGaMe07.pdf">original HyperLogLog paper</a> provides details the algorithm. The same authors released a <a href="http://cscubs.cs.uni-bonn.de/2016/proceedings/paper-03.pdf">paper</a> with a clearer view of the algorithm. Google released a <a href="https://research.google.com/pubs/pub40671.html">paper</a> containing a modified version named HyperLogLog+.```